x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java

```diff
@@ -18,7 +18,7 @@ static DeprecationIssue checkShardLimit(ClusterState state) {
         int maxShardsInCluster = shardsPerNode * nodeCount;
         int currentOpenShards = state.getMetaData().getTotalOpenIndexShards();
 
-        if (currentOpenShards >= maxShardsInCluster) {
+        if (nodeCount != 0 && currentOpenShards >= maxShardsInCluster) {
```

Comment:
This is a fix for bug that was revealed by the test for the new cluster-level deprecation check, but I sincerely doubt it would ever be hit in practice and I didn't think it was worth its own PR.

Reply:
nit: IMO `nodeCount > 0 && ...` reads better, else i am asking myself how this value can ever be negative.